### PR TITLE
describe the package properly, add the standard files

### PR DESCRIPTION
**Pull request by amiloradovsky**

### CODE REVIEW

File: meta.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+- name: metric spaces
+- name: real analysis
+- name: Urysohn's lemma
+- name: Tietze extension theorem
+
+categories:
+- name: Mathematics/Real Calculus and Topology
+
+documentation: |
+  Contents, roughly grouped in related categories:
```
Comment (Zimmi48) [inline meta.yml]:
```suggestion
  ## Contents, roughly grouped in related categories:
```

File: meta.yml [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+  # Basic definitions
```
Comment (Zimmi48) [inline meta.yml]:
```suggestion
  ### Basic definitions
```

and same for the following sub-sections.

File: meta.yml [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  - `RationalsInReals.v`
+  - `RTopology.v` - definition and properties of topology on R
+  - `RFuncContinuity.v` - reproof of continuity of basic functions on R
+
+  # "First nontrivial results of topology"
+
+  - `UrysohnsLemma.v`
+  - `TietzeExtension.v`
+
+  # Copyright
```
Comment (Zimmi48) [inline meta.yml]:
```suggestion
  ## Copyright
```

**Review by Zimmi48 — COMMENTED** [review]
Some other minor comments. I'll try to address the issues of the dependency in the CI today or tomorrow.

**Comment by Zimmi48** [comment]
Currently testing a fix at https://travis-ci.com/Zimmi48/topology/builds/102855363

**Comment by amiloradovsky** [comment]
```
trying https://github.com/coq-contribs/zorns-lemma/archive/719915f98c1da56ada31b058851a67b22788199a.tar.gz
unpacking source archive /tmp/nix-build-source.drv-0/719915f98c1da56ada31b058851a67b22788199a.tar.gz
building '/nix/store/873j17l2xlcrj0nm45xd13m6nld7x40n-coq8.6-zorns-lemma-v8.8.0.drv'...
```
But that commit (`719915f98c1da56ada31b058851a67b22788199a`) doesn't include the changes needed for it to compile (in my case at least), although tagged `v8.8`: at least version `6ac9bb914f6017cdd9a544ff4b0bef73fd33b44c` is needed.

**Comment by Zimmi48** [comment]
I updated the `zorns-lemma` package to the latest commit in the repo, but it means I had to "invent" a version number. It would be good to do a proper release soon (probably after merging the `meta.yml` PR though), and to publish the new release on opam.

**Comment by Zimmi48** [comment]
I've updated [my branch](https://github.com/Zimmi48/topology/tree/meta.yml). It should be OK now. I can push here if you want.

**Comment by amiloradovsky** [comment]
I haven't actually tested it on Coq 8.6, only 8.7, and just expected it to work there too. I suspect, fixing the build for that version wouldn't be difficult, but is probably not worth it now. Or is it?
Generally, yes, I imagined it as releasing `zorns-lemma` on OPAM with the latest changes and pointing the latest `topology` to it.
Sure you can push it here, I don't have plans to do something radically different.

**Comment by Zimmi48** [comment]
Coq 8.7 introduced several changes in the Reals library:

>IZR (Reals) has been changed to produce a compact representation of integers. As a consequence, IZR is no longer convertible to INR and lemmas such as INR_IZR_INZ should be used instead.
>
>Real constants are now represented using IZR rather than R0 and R1; this might cause rewriting rules to fail to apply to constants.

(extracts from CHANGES)

This can make it a bit difficult to provide compatibility with both 8.6 and 8.7 and is probably not worth it anyway since Coq 8.7.0 has been released one and a half year ago.

**Comment by Zimmi48** [comment]
FTR the best way to fix these error messages:

```
# File "./UrysohnsLemma.v", line 119, characters 11-14:
# Error: The reference Zlt was not found in the current environment.
# 
```

is to look at the warnings in 8.9 (possibly even making them errors) because they tell you with what to replace the deprecated notation.

**Comment by herbelin** [comment]
As part of a global update of coq-contribs to Coq 8.9 (and later to 8.10), I now have the possibility to make a release of topology compatible with Coq 8.9. How to proceed?

Is it ok if I push myself a V8.9.0 tag on the archive, what will automatically produce a 8.9 release? (Note that I actually already did [that](https://github.com/coq-community/zorns-lemma/releases/tag/v8.9.0) for ZornsLemma).

A possibility is also that I become the maintainer, and then, I can merge the current PR.

**Comment by Zimmi48** [comment]
@herbelin This PR is only about template files (README, testing, etc.). All the required compatibility fixes are already in master. In particular, you pushed a `v8.9` branch with commits that duplicated work that was already done (in master). It would have been probably better to ask first. I see you've done the same on the "semantics" project (with again duplicated work that had already been done in master). The model you have kept in mind from coq-contribs, and applied there, is not the preferred one for Coq libraries, where it is rather recommended having a single master branch which keeps compatibility with several successive versions of Coq at the same time, which is checked by CI. This removes any use for pushing a new branch every time there is a Coq release. See also: https://github.com/coq-community/manifesto/wiki/Commit,-branch-and-release-policies

**Comment by herbelin** [comment]
@zimmi48: I'm very sorry. zorns-lemma and topology were still in my archive of coq-contribs, so I upgraded them at the same time as the other coq-contribs without paying attention that there were other commits around. I'll consider the work done in coq-community to be the reference.

Regarding zorns-lemma and topology,  we should definitely decide if we remove them from coq-contribs or not. In the latter case, I'll not care anymore about upgrading them. For zorns-lemma and topology, there is still the issue that they don't have a well-identified "release" ensured to work with Coq 8.9 or 8.10, and in particular no opam package for Coq >= 8.9.

Regarding semantics, I initially forgot there were commits upstream but even after realizing that some job had been made, the resulting master was not compatible with v8.9, so I decided to keep a v8.9 branch anyway. Anyway, the duplication of work is lost of time only on my side. The work from contributors has been preserved in master!

> This removes any use for pushing a new branch every time there is a Coq release. 

I don't think new branches are useful, indeed. A tag would be enough if we can identify a time of the linear history where the package compile with the given version.

>  which is checked by CI.

Sorry for my ignorance, but I failed to see where to check CI compatibility for a coq-community package.

**Comment by Zimmi48** [comment]
@herbelin I hope I didn't sound harsh, it was not my intention to blame you in any way, and I'm indeed aware that the time lost is on your side. Sorry that I forgot to remove the submodules for these former contribs, as this is what led to the duplication.

To answer your question regarding CI, this is the point of the badges that are added in the README produced by the template. But not all projects in coq-community have adopted them yet (and semantics for instance has already a standard README but no CI setup yet).

**Comment by amiloradovsky** [comment]
I guess I now should just merge it as-is, despite some failed tests, and open new PRs/issues as needed. And then update the package(s) in Opam, but it wouldn't be wise from me to promise any short terms…

**Pull request merged**